receiver/filelogreceiver/filelog_test.go

```diff
@@ -168,16 +197,23 @@ func TestReadRotatingFiles(t *testing.T) {
 }
 
 type rotationTest struct {
-	name         string
-	copyTruncate bool
-	sequential   bool
+	name             string
+	copyTruncate     bool
+	sequential       bool
+	enableThreadPool bool
 }
 
 func (rt *rotationTest) Run(t *testing.T) {
 	t.Parallel()
 
 	tempDir := t.TempDir()
 
+	if rt.enableThreadPool {
+		t.Cleanup(func() {
+			require.NoError(t, featuregate.GlobalRegistry().Set("filelog.useThreadPool", false))
+		})
+		require.NoError(t, featuregate.GlobalRegistry().Set("filelog.useThreadPool", true))
+	}
```

Comment:
Does this test require its own management of the gate? Isn't it covered in `TestMain`?

Reply:
yes because both of them are separate packages. TestMain will only cover the `fileconsumer` package.